opentracing_instrumentation/client_hooks/requests.py

```diff
@@ -25,44 +25,53 @@
 import logging
 import urllib.parse
 
+from opentracing.ext import tags
 from ..request_context import get_current_span
 from ..http_client import AbstractRequestWrapper
 from ..http_client import before_http_request
 from ..http_client import split_host_and_port
-from ._singleton import singleton
+from ._patcher import Patcher
 
 log = logging.getLogger(__name__)
 
 # Try to save the original entry points
 try:
-    import requests.sessions
     import requests.adapters
 except ImportError:  # pragma: no cover
     pass
 else:
     _HTTPAdapter_send = requests.adapters.HTTPAdapter.send
 
 
-@singleton
-def install_patches():
-    try:
-        import requests.sessions
-        import requests.adapters
-    except ImportError:  # pragma: no cover
-        return
-
-    def send_wrapper(self, request, **kwargs):
-        """Wraps HTTPAdapter.send"""
-
-        request_wrapper = RequestWrapper(request=request)
-        span = before_http_request(
-            request=request_wrapper,
-            current_span_extractor=get_current_span)
-        with span:
-            resp = _HTTPAdapter_send(self, request, **kwargs)
-            if hasattr(resp, 'status_code') and resp.status_code is not None:
-                span.set_tag('http.status_code', resp.status_code)
-        return resp
+class RequestsPatcher(Patcher):
+
+    applicable = '_HTTPAdapter_send' in globals()
+    response_handler_hook = None
+
+    def set_response_handler_hook(self, response_handler_hook):
+        self.response_handler_hook = response_handler_hook
+
+    def _install_patches(self):
+        requests.adapters.HTTPAdapter.send = self.get_send_wrapper()
+
+    def _reset_patches(self):
+        requests.adapters.HTTPAdapter.send = _HTTPAdapter_send
+
+    def get_send_wrapper(self):
```

Comment:
nit: make it private, `_get_send_wrapper`

Reply:
It's done.